pkg/virt-handler/domain.go

```diff
@@ -48,14 +52,38 @@ func (d *DomainDispatch) Execute(indexer cache.Store, queue workqueue.RateLimiti
 		logging.DefaultLogger().Info().Object(domain).Msgf("Domain deleted")
 	} else {
 		domain = obj.(*virtwrap.Domain)
-		logging.DefaultLogger().Info().Object(domain).Msgf("Domain is in state %s", domain.Status.Status)
+		logging.DefaultLogger().Info().Object(domain).Msgf("Domain is in state %s reason %s", domain.Status.Status, domain.Status.Reason)
 	}
 	obj, vmExists, err := d.vmStore.GetByKey(key.(string))
 	if err != nil {
 		queue.AddRateLimited(key)
 	} else if !vmExists || obj.(*v1.VM).GetObjectMeta().GetUID() != domain.GetObjectMeta().GetUID() {
 		// The VM is not in the vm cache, or is a VM with a differend uuid, tell the VM controller to investigate it
 		d.vmQueue.Add(key)
+	} else {
+		setVmPhaseForStatusReason(domain, obj.(*v1.VM), d.restClient)
```

Comment:
You also have to check if the VM has the same uid like the Domain, otherwise you are setting a potentially new posted VM to stopped, although it was not even scheduled ...